torchvision/csrc/cpu/video/Video.cpp

```diff
@@ -0,0 +1,393 @@
+      };
+    }
+  }
+  streamFPS.insert({{"video", videoFPS}, {"audio", audioFPS}});
+  streamDuration.insert({{"video", videoDuration}, {"audio", audioDuration}});
+
+  succeeded = Video::_setCurrentStream();
+  LOG(INFO) << "\nDecoder inited with: " << succeeded << "\n";
+  if (get<1>(current_stream) != -1) {
+    LOG(INFO)
+        << "Stream index set to " << get<1>(current_stream)
+        << ". If you encounter trouble, consider switching it to automatic stream discovery. \n";
+  }
+} // video
+
+bool Video::_setCurrentStream() {
+  double ts = 0;
+  if (seekTS > 0) {
+    ts = seekTS;
+  }
+
+  _getDecoderParams(
+      ts, // video start
+      0, // headerOnly
+      get<0>(current_stream), // stream
+      long(get<1>(
+          current_stream)), // stream_id parsed from info above change to -2
+      false // read all streams
+  );
+
+  // calback and metadata defined in Video.h
+  return (decoder.init(params, std::move(callback), &metadata));
+}
+
+std::tuple<std::string, int64_t> Video::getCurrentStream() const {
+  return current_stream;
+}
+
+std::vector<double> Video::getFPS(std::string stream) const {
+  // add safety check
+  if (stream.empty()) {
+    stream = get<0>(current_stream);
+  }
+  auto stream_tpl = _parseStream(stream);
+  std::string stream_str = get<0>(stream_tpl);
+  // check if the stream exists
+  return streamFPS.at(stream_str);
+}
+
+std::vector<double> Video::getDuration(std::string stream) const {
+  // add safety check
+  if (stream.empty()) {
+    stream = get<0>(current_stream);
+  }
+  auto stream_tpl = _parseStream(stream);
+  std::string stream_str = get<0>(stream_tpl);
+  // check if the stream exists
+  return streamDuration.at(stream_str);
+}
+
+void Video::Seek(double ts, bool any_frame = false) {
+  // initialize the class variables used for seeking and retrurn
+  video_any_frame = any_frame;
+  seekTS = ts;
+  doSeek = true;
+}
+
+torch::List<torch::Tensor> Video::Next(std::string stream) {
+  
+  bool newInit = false; // avoid unnecessary decoder initializations
+  if ((!stream.empty()) && (_parseStream(stream) != current_stream)) {
+      current_stream = _parseStream(stream);
+      newInit = true;
+  }
+
+  if ((seekTS != -1) && (doSeek == true)) {
+      newInit = true;
+      doSeek = false;
+  }
+
+  if (newInit){
+    succeeded = Video::_setCurrentStream();
+    if (succeeded) {
+      newInit = false;
+    }
+  }
+
+  // if failing to decode simply return a null tensor (note, should we
+  // raise an exeption?)
+  torch::Tensor framePTS = torch::zeros({1}, torch::kFloat);
+  torch::Tensor outFrame = torch::zeros({0}, torch::kByte);
+
+  // decode single frame
+  DecoderOutputMessage out;
+  int64_t res = decoder.decode(&out, decoderTimeoutMs);
+  // if successfull
+  if (res == 0) {
+    auto header = out.header;
+    const auto& format = header.format;
+
+    // initialize the output variables based on type
+    size_t expectedWrittenBytes = 0;
```

Comment:
Is this used for something? Do you want to use this to check that we wrote the same number of bytes? We pass a torch Tensor around, probably not needed to keep this around